Sources/JavaScriptKit/FundamentalObjects/JSFunction.swift

```diff
@@ -93,11 +93,11 @@ internal func invokeJSFunction(_ jsFunc: JSFunction, arguments: [ConvertibleToJS
             var payload1 = JavaScriptPayload1()
             var payload2 = JavaScriptPayload2()
             if let thisId = this?.id {
-                _call_function_with_this(thisId,
+                _call_function_with_this_unsafe(thisId,
```

Comment:
`invokeJSFunction` is also used by `JSThrowingFunction`, so I think you’d have to copy-paste the code and update it as appropriate.

I think the non-throwing version should do an `assert(!kindAndFlags.isException` before simply returning the `JSValue`:

```swift
internal func invokeNonThrowingJSFunction(_ jsFunc: JSFunction, arguments: [ConvertibleToJSValue], this: JSObject?) throws -> JSValue {
    rawValues.withUnsafeBufferPointer { bufferPointer -> (JSValue, Bool) in
        let argv = bufferPointer.baseAddress
        let argc = bufferPointer.count
        var kindAndFlags = JavaScriptValueKindAndFlags()
        var payload1 = JavaScriptPayload1()
        var payload2 = JavaScriptPayload2()
        if let thisId = this?.id {
            _call_function_with_this_unsafe(
                thisId,
                jsFunc.id, argv, Int32(argc),
                &kindAndFlags, &payload1, &payload2
            )
        } else {
            _call_function_unsafe(
                jsFunc.id, argv, Int32(argc),
                &kindAndFlags, &payload1, &payload2
            )
        }
        assert(!kindAndFlags.isException)
        let result = RawJSValue(kind: kindAndFlags.kind, payload1: payload1, payload2: payload2)
        return result.jsValue()
    }
}

Reply:
Thanks! I've applied the change already in a new commit 😃